src/hotspot/share/opto/subnode.cpp

```diff
@@ -1839,6 +1839,72 @@ bool BoolNode::is_counted_loop_exit_test() {
   return false;
 }
 
+//=============================================================================
+//------------------------------Value------------------------------------------
+const Type* AbsNode::Value(PhaseGVN* phase) const {
+  const Type* t1 = phase->type(in(1));
+  if (t1 == Type::TOP) return Type::TOP;
+
+  switch (t1->base()) {
+  case Type::Int: {
+    const TypeInt* ti = t1->is_int();
+    if (ti->is_con()) {
+      // Special case for min_jint: Math.abs(min_jint) = min_jint.
+      // Do not use C++ abs() for min_jint to avoid undefined behavior.
+      return (ti->is_con(min_jint)) ? TypeInt::MIN : TypeInt::make(abs(ti->get_con()));
```

Comment:
```suggestion
      return TypeInt::make(uabs(ti->get_con());
```
We have uabs() for julong and unsigned int.

Reply:
Thanks for your review. Fixed.